components/style/properties/declaration_block.rs

```diff
@@ -102,12 +99,14 @@ impl PropertyDeclarationBlock {
     /// Vec::extend
     pub fn extend(&mut self, other: PropertyDeclarationBlock) {
         self.declarations.extend(other.declarations);
-        self.important_count += other.important_count
+        self.important_count += other.important_count;
+        self.is_deduplicated &= other.is_deduplicated;
```

Comment:
Is this bit correct? What guarantees that `other` doesn't contain duplicated declarations with `self`?